### ISSUE TITLE
OrthoFinder Algorithm failed with error

### ISSUE THREAD

**Issue by Pleetzken** [davidemms/OrthoFinder#276]
Hi,
My Orthofinder run with 102 species aborted with the below error. Got the error after adding 10 species to 92 species and then after restarting from the blast results of the first try of 102 species both times the same error during OrthoFinder algorithm. I tried to find any irregularities in the input data, but found none. 
Thanks for any help or directions, you can give
Peter

orthofinder output: (slighlty shortened at [...])

```
OrthoFinder version 2.3.3 Copyright (C) 2014 David Emms

2019-07-18 10:18:37 : Starting OrthoFinder
28 thread(s) for highly parallel tasks (BLAST searches etc.)
1 thread(s) for OrthoFinder algorithm

Checking required programs are installed
----------------------------------------
Test can run "mcl -h" - ok
Test can run "fastme [...]" - ok
Using previously calculated BLAST results in [...]/Results_add_new_specs1/WorkingDirectory/

Running OrthoFinder algorithm
-----------------------------
Traceback (most recent call last):
  File "/usr/local/sbin/orthofinder", line 1661, in <module>
    statsFile, summaryText, orthogroupsResultsFilesString = DoOrthogroups(options, speciesInfoObj, seqsInfo)    
  File "/usr/local/sbin/orthofinder", line 1277, in DoOrthogroups
    Lengths = GetSequenceLengths(seqsInfo)
  File "/usr/local/sbin/orthofinder", line 357, in GetSequenceLengths
    _, iCurrentSequence = util.GetIDPairFromString(row[1:])
  File "/home/peter/Downloads/OrthoFinder-master/orthofinder/scripts/util.py", line 342, in GetIDPairFromString
    return map(int, line.split("_"))
ValueError: invalid literal for int() with base 10: 'XP'

```

**Comment by Pleetzken** [comment]
Hi David,
thank you for your answer. The output from `grep ">" Species*fa | grep XP` is empty (as is `grep ">" Species*fa | grep "[[A-Za-z]]` ).
I added the new species with
`orthofinder -f add_new_specs/protein/ -b old_specs/OrthoFinder/Orthofinder_Results/ -t 28 -n add_new_specs1`
The irregular thing I did was:
- For the original 92 species I added them step by step with the `-f -b` combination
- Then I manipulated the last WorkingDirectory so it would work without all previous Working Directories before it:
I deleted the previousWD file, changed the path in the log file, and copied all Blast* files and Species*fa files from the previousWDs to the last one

Maybe this was my mistake. If so, is there a way to achieve what I tried, that is taking only the last Results from a cumulative run for further runs?

Best
Peter

**Comment by davidemms** [comment]
Hi Peter 

I wonder if after the changes to the directories OrthoFinder is looking in the wrong place for files. Are you happy making a small edit to the code to print out where it's getting the file from? In the output you posted it said the error was here:
```
  File "/usr/local/sbin/orthofinder", line 357, in GetSequenceLengths
    _, iCurrentSequence = util.GetIDPairFromString(row[1:])
```
so a few lines above this, you could insert a print statement so it says
```
        qFirstLine = True
        print(fastaFilename)                            #  inserted line
        with open(fastaFilename) as infile:
```

The will tell you what files are being opened by this code and the last filename printed before the error will be the one causing the problem. The files it _should_ be opening are the Speices*.fa files in the WorkingDirectory that it prepared and all the accessions should be the IDs described above. Let me know how you get on.

All the best
David

**Comment by Pleetzken** [comment]
Hi David,
I made the change, and checked the files. 
I have to apologize, since I found your first answer was the problem and I looked in the wrong WorkingDirectory folder(in the one with the 10 new seqs, not the one with the 92 old ones). In the old WorkingDirectory, there were indeed the Species*.fa files with the original sequence names (>XP_*...) so I now changed them with the files with the correct sequence names (>0_0 etc) and it seems to be working now. I must have mixed something up, when I copied the files.
So thank you very much for your help
Peter Lyko
